AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Storage/StorageEngine.swift

```diff
@@ -415,6 +422,43 @@ final class StorageEngine: StorageEngineBehavior {
         }
     }
 
+    @available(iOS 13.0, *)
+    private func syncDeletions(of associatedModels: [Model],
+                               syncEngine: RemoteSyncEngineBehavior,
+                               completion: @escaping DataStoreCallback<Void>) {
+        guard !associatedModels.isEmpty else {
+            completion(.successfulVoid)
+            return
+        }
+        var mutationEvents: Set<Model.Identifier> = []
```

Comment:
1. Any reason we're not just using a counter of successful mutation events?
2. Is there a data race here, or is the callback to `submitToSyncEngine` guaranteed to always execute on the same serial DispatchQueue, so that this Set is not modified except on that queue?